src/stacks-hierarchy/Root_PurchasePaymentScreen/Root_PurchasePaymentScreen.tsx

```diff
@@ -51,6 +52,20 @@ export const Root_PurchasePaymentScreen = ({route, navigation}: Props) => {
     reserveMutation.data?.recurring_payment_id,
   );
 
+  useEffect(() => {
+    const orderId = reserveMutation.data?.order_id;
+    if (orderId) {
+      const fareContract = [...(fareContracts ?? []), ...(sentFareContracts ?? [])].map((fc) => {
+        return {
+          orderId: fc.orderId,
+        };
+      }).find((contract) => contract.orderId === orderId);
```

Comment:
This can be simplified by using `.some`
```ts
const allPossibleFareContracts = [...(fareContracts ?? []), ...(sentFareContracts ?? [])]
const fareContractReceived = allPossibleFareContracts.some((fc) => fc.orderId === orderId);
```